galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreValidator.java

```diff
@@ -9,23 +9,198 @@
+    }
+
+    private void tryToCheckAndUpdateCouchDBTokenView(URI couchdbUri, CloseableHttpClient httpClient,
+            HttpRequestFactory httpRequestFactory) throws CouchdbException {
         // Perform the base CouchDB checks
         super.checkCouchdbDatabaseIsValid(couchdbUri, httpClient, httpRequestFactory);
-
         validateDatabasePresent(couchdbUri, CouchdbAuthStore.TOKENS_DATABASE_NAME);
+        checkTokensDesignDocument(httpClient, couchdbUri, 1);
+    }
 
-        logger.debug("Auth Store CouchDB at " + couchdbUri.toString() + " validated");
+    public void checkTokensDesignDocument(CloseableHttpClient httpClient, URI couchdbUri, int attempts)
+            throws CouchdbException {
+
+        // Get the design document from couchdb
+        String docJson = getTokenDesignDocument(httpClient, couchdbUri, attempts);
+
+        TokensDBNameViewDesign tableDesign = parseTokenDesignFromJson(docJson);
+
+        boolean isDesignUpdated = transformDesignToDesired(tableDesign);
+
+        if (isDesignUpdated) {
+            updateTokenDesignDocument(httpClient, couchdbUri, attempts, tableDesign);
+        }
+    }
+
+    private TokensDBNameViewDesign parseTokenDesignFromJson(String docJson) throws CouchdbException {
+        TokensDBNameViewDesign tableDesign;
+        try {
+            tableDesign = gson.fromJson(docJson, TokensDBNameViewDesign.class);
+        } catch (JsonSyntaxException ex) {
+            throw new CouchdbException(ERROR_FAILED_TO_PARSE_COUCHDB_DESIGN_DOC.getMessage(ex.getMessage()), ex);
+        }
+
+        if (tableDesign == null) {
+            tableDesign = new TokensDBNameViewDesign();
+        }
+        return tableDesign;
+    }
+
+    private boolean transformDesignToDesired(TokensDBNameViewDesign tableDesign) {
+        boolean isUpdated = false;
+
+        if (tableDesign.views == null) {
+            isUpdated = true;
+            tableDesign.views = new TokenDBViews();
+        }
+
+        if (tableDesign.views.loginIdView == null) {
+            isUpdated = true;
+            tableDesign.views.loginIdView = new TokenDBLoginView();
+        }
+
+        if (tableDesign.views.loginIdView.map == null
+                || !DB_TABLE_TOKENS_DESIGN.equals(tableDesign.views.loginIdView.map)) {
+            isUpdated = true;
+            tableDesign.views.loginIdView.map = DB_TABLE_TOKENS_DESIGN;
+        }
+
+        if (tableDesign.language == null || !tableDesign.language.equals("javascript")) {
+            isUpdated = true;
+            tableDesign.language = "javascript";
+        }
+
+        return isUpdated;
+    }
+
+    private String getTokenDesignDocument(CloseableHttpClient httpClient, URI couchdbUri, int attempts)
+            throws CouchdbException {
+        HttpRequestFactory requestFactory = super.getRequestFactory();
+        HttpGet httpGet = requestFactory.getHttpGetRequest(couchdbUri + "/galasa_tokens/_design/docs");
```

Comment:
Minor suggestion - using a constant (`CouchdbAuthStore.TOKENS_DATABASE_NAME` already exists) for the tokens database name would be good instead of hardcoding `galasa_tokens` to avoid typos - same applies throughout this file